be/src/storage/lake/horizontal_compaction_task.cpp

```diff
@@ -58,6 +58,9 @@ Status HorizontalCompactionTask::execute(Stats* stats) {
     reader_params.chunk_size = chunk_size;
     reader_params.profile = nullptr;
     reader_params.use_page_cache = false;
+    if (keys_type == PRIMARY_KEYS) {
+        reader_params.sorted_by_keys_per_tablet = true;
```

Comment:
Try to handle primary key and duplicate key in the same way at the same place:
https://github.com/StarRocks/starrocks/blob/af7d5632703ef2887dc6114b359112fa20a4152b/be/src/storage/lake/tablet_reader.cpp#L250

Reply:
Done